File: Program51.java
import java.lang.*;
import java.util.*;

class Digits
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number");
        this.iNo = sobj.nextInt();
    }
    public void Display()
    {
        System.out.println("Entered value is: "+this.iNo);
    }
    public boolean CheckPallindrome()
    {
        int iDigit = 0;
        int iRev = 0;
        int iTemp = 0;
        
        iTemp = iNo;
        
        while(iNo != 0)
        {
            iDigit = iNo % 10;
            iRev = iRev * 10 + iDigit;
            iNo = iNo / 10;
        }
        
        if(iRev == iTemp)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

class Program51
{
    public static void main(String arg[])
    {
        boolean bRet;

        Digits dobj = new Digits();

        dobj.Accept();
        dobj.Display();
        
        bRet = dobj.CheckPallindrome();
        if(bRet == true)
        {
            System.out.println("Pallindrome number");
        }
        else
        {
            System.out.println("Not Pallindrome number");
        }
    }
}

File: Program51.c
#include<stdio.h>
#include<stdbool.h>

////////////////////////////////////////////////////////////////
//
// Function Name:        
// Description:          Display digits in reverse order
// Input:                Integer
// Output:               Integer
// Author Name:          Sursj Rajkumar More
//
/////////////////////////////////////////////////////////////////

bool CheckPallindrome(int iNo)
{
   int iDigit = 0;
   int iRev = 0;
   int iTemp = 0;

   iTemp = iNo;

   while(iNo != 0)
   {
       iDigit = iNo % 10;
       iRev = iRev * 10 + iDigit;
       iNo = iNo / 10;
   }
   if(iRev == iTemp)
   {
       return true;
   }
   else
   {
       return false;
   }
}

//////////////////////////////////////////////////////////////////////////////
// Write a program to display digits in reverse order
/////////////////////////////////////////////////////////////////////////////

int main()
{
    int iValue = 0;
    bool bRet;

    printf("Enter number\n");
    scanf("%d",&iValue);

    bRet = CheckPallindrome(iValue);
    
    if(bRet == true)
    {
        printf("%d is a pallindrome number",iValue);
    }
    else
    {
        printf("%d is not a pallindrome number",iValue);
    }
    
    return 0;
}

////////////////////
//
// Input:    123
// Output:   321
//
///////////////////